Adds notes and updates some components from component audit part 2

jenniferthibault opened the pull request:
## Summary
This makes many of the simple component updates discovered in the audit from #67, and adds notes to the existing components. 

✅ Icon headings: 
 - Adds new examples and notes

✅ Option components:
- Adds more component examples in a row, and adds notes

✅ Overlay:
- Changes name to Loading overlay, adds notes

✅ Reaction box:
- Adds notes

✅ Side nav:
- Changes category name to "Side navigation"
- Changes the name of each component to clarify its function. Adds notes and examples to each

✅ Sidebar:
- Changes name to Resource bar
- Adds two sub-categories, adds notes and examples to each

✅ Slab:
- Shows all color variations on one page to be more consistent with button examples, adds notes

✅ Tooltip:
- Adds notes

⚠️ Filter tags:
- Adds examples of filters in grids, and adds notes
  - I ran into a quirk I didn't know how to solve here, where the separators (or's, hyphens) dropped out of the tag line. Would appreciate some review help on this especially!
![screen shot 2018-03-03 at 2 05 57 am](https://user-images.githubusercontent.com/11636908/36957154-7171f87e-2000-11e8-8e1a-eb9b24246944.png)

xtine commented on components/02-components/reaction-box/README.md:
```diff
@@ -1 +1,8 @@
-*There are no notes for this component yet.* 
+A reaction box prompts users to give feedback about a specific new feature in the context of the production site.
```
Should we add some information about why it has a "NOT IN USE" label on it?

xtine commented on components/02-components/secondary-navigation/01-side-navigation/README.md:
```diff
@@ -0,0 +1,17 @@
+A hierarchical vertical navigation on the left side of a page that helps users navigate between pages within a section of the website.
```
I think it would be helpful to expand on navigation -- that the links on the side nav actually click out to new pages and doesn't do a non-refreshed (ajax) load and not to be confused with the other panels that jump to sections on content on the same page.

xtine commented on components/02-components/tooltips/README.md:
```diff
@@ -1 +1,12 @@
+
+## When to use
+- When a setting or control is highly specific to the FEC’s unique subject matter or context </br>
+_Ex: The “[Restrict contributions](https://www.fec.gov/data/receipts/?data_type=processed&two_year_transaction_period=2018&min_date=01%2F01%2F2017&max_date=03%2F02%2F2018)” filter_
```
For some reason the single underscore italic isn't rendering on the notes section.

jenniferthibault commented:
Thanks for the review and feedback! I made the following changes to addres your comments:
- Added more information on the "not in use" status, and connected a relevant github issue to the history of the design
- Clarified when the side navigation component triggers a page load (I tried to keep it not too technical, but maybe it need to be moreso ?)
- Fixed a link so that markdown would recognize the italicized stye
- Added clarification around the CSS hack note to specify that it's only a problem for the pattern library instance

Take a peek, and let me know if there are any adjustments needed, or feel free to merge away!

xtine commented on components/02-components/secondary-navigation/01-side-navigation/README.md:
```diff
@@ -1,17 +1,17 @@
-A hierarchical vertical navigation on the left side of a page that helps users navigate between pages within a section of the website.
+A hierarchical vertical navigation on the left side of a page that helps users navigate between pages within a section of the website. Top-level links navigate to new pages, and initiate a new page load. Second-level links navigate to sections within that page.
```
I like this explanation!
